experimental/framework/fluid-static/src/types.ts

```diff
@@ -71,3 +71,75 @@ export interface ContainerSchema {
      */
     dynamicObjectTypes?: LoadableObjectClass<any>[];
 }
+
+/**
+ * Base interface to be implemented to fetch each service's audience. The generic C allows consumers to further
+ * extend the client object with service-specific details about the connecting client, such as device information,
+ * environme
+ */
+export interface IServiceAudience<M extends IMember> {
+    /**
+     * Returns an map of all users currently in the Fluid session where key is the userId and the value is the
+     * member object
+     */
+    getMembers(): Map<string, M>;
+
+    /**
+     * Returns the current active user on this client once they are connected. Otherwise, returns undefined.
+     */
+    getCurrentMember(): M | undefined;
+
+    /**
+     * Returns the current client's details once it is connected. Otherwise, returns undefined.
+     */
+    getCurrentClient(): IConnectedClient | undefined;
+
+    /**
+     * Returns the member who made the last edit along with the edit's timestamp
+     */
+    getLastEdited(): ILastEditedResult<M> | undefined;
```

Comment:
It seems difficult to scale/scope the duties of the audience if it also needs to contain accessory functionality like last-edited (e.g. as people want new information about the audience, do we have to keep expanding this interface?).

Is there some way we can have the last edited information live separately from the audience but be capable of correlating with audience members (e.g. through a unique ID or something)?

Reply:
As we chatted about, I've removed the default consumption of the last-edited-experimental data object from the container. I think we can have that be separately added in by the user through the registry, and they could use the userId returned by the audience to separately correlate the values returned by last-edited to the members returned by the audience